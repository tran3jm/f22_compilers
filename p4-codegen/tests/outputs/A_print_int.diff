--- outputs/A_print_int.txt	2022-11-16 21:18:35.258108591 -0500
+++ expected/A_print_int.txt	2022-11-08 11:50:27.626554085 -0500
@@ -1,13 +1 @@
-main:
-  push BP
-  i2i SP => BP
-  addI SP, 0 => SP
-  print ""
-  loadI 0 => r1
-  i2i r1 => RET
-  jump l0
-l0:
-  i2i BP => SP
-  pop BP
-  return
-RETURN VALUE = 0
+7RETURN VALUE = 0
