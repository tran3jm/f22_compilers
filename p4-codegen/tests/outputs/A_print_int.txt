main:
  loadI 0 => r1
  i2i r1 => RET
l0:
  return

==========================
sp=65536 bp=-9999999 ret=-9999999
virtual regs: 
stack:
other memory:
==========================

Executing: loadI 0 => r1

==========================
sp=65536 bp=-9999999 ret=-9999999
virtual regs:  r1=0
stack:
other memory:
==========================

Executing: i2i r1 => RET

==========================
sp=65536 bp=-9999999 ret=0
virtual regs:  r1=0
stack:
other memory:
==========================

Executing: l0:

==========================
sp=65536 bp=-9999999 ret=0
virtual regs:  r1=0
stack:
other memory:
==========================

Executing: return
RETURN VALUE = 0
