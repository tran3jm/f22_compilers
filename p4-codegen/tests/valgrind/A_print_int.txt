==3471010== Memcheck, a memory error detector
==3471010== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==3471010== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==3471010== Command: ../decaf inputs/print_int.decaf
==3471010== 
Error: ast.dot: syntax error in line 3 near ''
main:
  push BP
  i2i SP => BP
  addI SP, 0 => SP
  print ""
  loadI 0 => r1
  i2i r1 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return
RETURN VALUE = 0
==3471010== 
==3471010== HEAP SUMMARY:
==3471010==     in use at exit: 0 bytes in 0 blocks
==3471010==   total heap usage: 1,820 allocs, 1,820 frees, 373,788 bytes allocated
==3471010== 
==3471010== All heap blocks were freed -- no leaks are possible
==3471010== 
==3471010== For lists of detected and suppressed errors, rerun with: -s
==3471010== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
