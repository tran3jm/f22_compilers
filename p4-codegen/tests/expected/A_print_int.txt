7RETURN VALUE = 0
